src/storage/repositories/zip.py

```diff
@@ -24,8 +25,19 @@ def update(self, entity: dict, storage_recipe=None, **kwargs):
     def get(self, uid: str):
         return "Not implemented on ZipFile repository!"
 
-    def add(self, uid: str, document: dict):
-        return "Not implemented on ZipFile repository!"
+    def add(self, document: dict, path: str, filename: str = None):
+        """Add the provided document as a  json file to the zip file"""
+        if path[-1] == "/":
+            raise ValidationException(message=f"When adding file to Zip, path ({path}) should not end with a '/'")
+        json_data = json.dumps(document)
+        binary_data = json_data.encode()
+
+        if filename:
+            write_path = f"{path}/{filename}.json"
+        else:
+            write_path = f"{path}/{document['name']}.json"
```

Comment:
Perhaps the caller should handle this, so this function always uses the filename?
Also, what happens if the document does not have a ["name"]?